core/runtime/src/main/java/io/quarkus/runtime/configuration/LocalConfigSource.java

```diff
@@ -0,0 +1,107 @@
+package io.quarkus.runtime.configuration;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.jboss.logging.Logger;
+
+import io.smallrye.config.EnvConfigSource;
+
+/**
+ * A configuration source which supports local config file.
+ */
+final public class LocalConfigSource implements ConfigSource {
```

Comment:
Since it's no longer a `.env` situation, maybe it makes more sense to make another nested class in `ApplicationPropertiesConfigSource`.  All the rewriting code down on https://github.com/quarkusio/quarkus/pull/4258/files#diff-6ba9ba821829318906022ab5610f5f7cR55 could be dropped really.